openpype/hosts/max/plugins/publish/extract_tycache.py

```diff
@@ -0,0 +1,169 @@
+
+        Those attributes are to be exported in MAX Script.
+
+        Args:
+            members (list): Member nodes of the instance.
+            start (int): Start frame.
+            end (int): End frame.
+            filepath (str): Output path of the TyCache file.
+
+        Returns:
+            list of arguments for MAX Script.
+
+        """
+        job_args = []
+        opt_list = self.get_operators(members)
+        for operator in opt_list:
+            if tycache_spline_enabled:
+                export_mode = f'{operator}.exportMode=3'
+                job_args.append(export_mode)
+            else:
+                export_mode = f'{operator}.exportMode=2'
+                job_args.append(export_mode)
+            start_frame = f"{operator}.frameStart={start}"
+            job_args.append(start_frame)
+            end_frame = f"{operator}.frameEnd={end}"
+            job_args.append(end_frame)
+            filepath = filepath.replace("\\", "/")
+            tycache_filename = f'{operator}.tyCacheFilename="{filepath}"'
+            job_args.append(tycache_filename)
+            # TODO: add the additional job args for tycache attributes
+            if additional_attributes:
+                additional_args = self.get_additional_attribute_args(
+                    operator, additional_attributes
+                )
+                job_args.extend(additional_args)
+            tycache_export = f"{operator}.exportTyCache()"
+            job_args.append(tycache_export)
+
+        return job_args
+
+    @staticmethod
+    def get_operators(members):
+        """Get Export Particles Operator.
+
+        Args:
+            members (list): Instance members.
+
+        Returns:
+            list of particle operators
+
+        """
+        opt_list = []
+        for member in members:
+            obj = member.baseobject
+        # TODO: to see if it can be used maxscript instead
+            anim_names = rt.GetSubAnimNames(obj)
+            for anim_name in anim_names:
+                sub_anim = rt.GetSubAnim(obj, anim_name)
+                boolean = rt.IsProperty(sub_anim, "Export_Particles")
+                if boolean:
+                    event_name = sub_anim.Name
+                    opt = f"${member.Name}.{event_name}.export_particles"
+                    opt_list.append(opt)
+
+        return opt_list
+
+    def get_additional_attribute_args(self, operator, attrs):
+        """Get Additional args with the attributes pre-set by user
+
+        Args:
+            operator (str): export particle operator
+            attrs (dict): a dict which stores the additional attributes
+            added by user
+
+        Returns:
+            additional_args(list): a list of additional args for MAX script
+        """
+        additional_args = []
+        for key, value in attrs.items():
+            tyc_attribute = None
+            if isinstance(value, bool):
+                tyc_attribute = f"{operator}.{key}=True"
```

Comment:
Regardless of the `bool` value, it will always be `True` ?

Reply:
it's only happened to be True when the user add the tycache parameter for publish in the publish tab.